### PR TITLE
[modernize][Videoplayer] Prefer 'override' usage

### PR DESCRIPTION
**Pull request by hudokkow**
Follow up of https://github.com/garbear/xbmc/pull/83

@garbear, @FernetMenta

### CODE REVIEW

**Comment by garbear** [comment]
I'll follow up with some changes from the comments made in #12250

File: xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemux.h [hunk 1]
```diff
@@ -243,29 +243,29 @@ class CDVDDemux
   /*
    * Reset the entire demuxer (same result as closing and opening it)
    */
-  virtual void Reset() = 0;
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemux.h]: I already mentioned that this is wrong. please don't change behaviour in such kind of bulk cosmetic changes.

**Comment by FernetMenta** [comment]
do not hide architectural changes into this kind of PRs. that is the reason why #12250 got stuck

**Comment by garbear** [comment]
@hudokkow please remove my comment from your PR. I sent the updated version to https://github.com/hudokkow/xbmc/pull/1

**Comment by garbear** [comment]
@FernetMenta I have addressed the architectural comments from #12250: https://github.com/hudokkow/xbmc/pull/1/files

Should I send to this PR, or wait for this PR to be merged (without any architectural changes) and then PR to `vpupdates`?

**Comment by FernetMenta** [comment]
please focus only on "override" changes in this PR. this is what the title says. every thing else we should do in separeate PRs.
btw @hudokkow thanks for doing this on this branch.

File: xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemux.h (hunk 1 above)
Comment (hudokkow) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemux.h]: Think this was some sort of rebase stupidity and lack of attention from my part. Oddly, after rebasing today it went away by itself.

**Comment by hudokkow** [comment]
Rebased and cleaned/fixed up a lot of vertical alignment stuff.

Better review with https://github.com/xbmc/xbmc/pull/12440/files?w=1 to ignore tab/space changes.

**Comment by hudokkow** [comment]
jenkins build this please

**Comment by FernetMenta** [comment]
thanks

**Pull request merged**